php/ui/class-component.php

```diff
@@ -762,19 +762,21 @@ public static function init( $setting ) {
 
 		$caller = get_called_class();
 		$type   = $setting->get_param( 'type' );
-		// Set Caller.
-		$component = "{$caller}\\{$type}";
 		// Final check if type is callable component.
 		if ( ! is_string( $type ) || ! self::is_component_type( $type ) ) {
-			// Set to a default HTML component if not found.
-			$type = 'html';
 			// Check what type this component needs to be.
 			if ( is_callable( $type ) ) {
 				$setting->set_param( 'callback', $type );
 				$setting->set_param( 'type', 'custom' );
 				$type = 'custom';
+			} else {
```

Comment:
Not quite relevant, but IMHO, I prefer how it was. Setting a default that can be conditionally overridden. As much as possible, I try to avoid `else` conditions. As I mentioned, this is more of a personal preference and not a change request.

Reply:
@pereirinha - there is a specific reason I did it this way, one which is not obvious. I can show you later in our sync.